server/sonar-db-dao/src/main/resources/org/sonar/db/qualityprofile/ActiveRuleMapper.xml

```diff
@@ -191,7 +191,9 @@
     select p.kee as "key", count(ar.id) as "value"
     from active_rules ar
     inner join rules_profiles p on p.id = ar.profile_id
+    inner join rules r on r.id = ar.rule_id
     where p.organization_uuid = #{organizationUuid, jdbcType=VARCHAR}
+    and r.status != 'REMOVED'
```

Comment:
A documentation could be added to `ActiveRuleDao#countActiveRulesByProfileKey()` and `#countActiveRulesForInheritanceByProfileKey()` to mention the fact that these rows are filtered out.